src/Metric/MetricFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Metric;
+
+use Baldinof\RoadRunnerBundle\Exception\UnknownRpcTransportException;
+use Spiral\Goridge\RPC;
+use Spiral\Goridge\SocketRelay;
+use Spiral\RoadRunner\Metrics;
+use Spiral\RoadRunner\MetricsInterface;
+
+class MetricFactory
+{
+    /**
+     * @var bool
+     */
+    private $metricsEnabled;
+
+    /**
+     * @var string
+     */
+    private $rrRpc;
+
+    /**
+     * @var string
+     */
+    private $kernelProjectDir;
+
+    public function __construct(string $rrRpc, string $kernelProjectDir, bool $metricsEnabled)
+    {
+        $this->rrRpc = $rrRpc;
+        $this->kernelProjectDir = $kernelProjectDir;
+        $this->metricsEnabled = $metricsEnabled;
+    }
+
+    public function getMetricService(): MetricsInterface
+    {
+        $nullMetrics = new NullMetrics();
+
+        if (!$this->metricsEnabled || !$this->rrRpc) {
```

Comment:
```suggestion
        if (!$this->metricsEnabled) {
```

`NullMetrics` should be used only when the user explicitly disabled metrics. 

Other cases are results of bad configuration, so we should also throw an exception if metrics are enabled but `$this->rrRpc === null` (likely the user forgot to enable RPC in RoadRunner).

Reply:
Fixed